### PR TITLE
improved generation of CHANGELOG.md

### PR DESCRIPTION
**Pull request by wesleymccollam**

### CODE REVIEW

File: scripts/generate-changelog.sh [hunk 1]
```diff
@@ -1,21 +1,41 @@
...
+    issue_number=$(echo "$pr_content" | grep $target_pr | awk -F "\t" '{print $3}' | awk -F "-" '{print $1}')
     issue_content=$(gh issue view $issue_number)
     issue_title_content=$(echo "$issue_content" | grep -E "title" | awk -F ": " '{print $NF}')
     issue_title=$(echo $issue_title_content | awk -F ": " '{print $NF}')
     issue_label_content=$(echo "$issue_content" | grep -E "labels")
-    issue_label=$(echo $issue_label_content | awk -F " " '{print $NF}')
+    issue_label=$(echo $issue_label_content | awk -F ": " '{print $NF}')
+    issue_label_count=$(echo "$issue_label" | sed -e 's/[,]//g' | wc -w | sed -e "s/ //g")
     
+    if [ $issue_label_count -gt 1 ] ; then
```
Comment (henryrecker-pingidentity) [inline scripts/generate-changelog.sh:23]: Why is it called conflict when there is more than one label?

File: contributing/release-notes.md [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+### *This documented process is to only be used by maintainers of this repository!*
 ## Prerequisites
 - [terraform-provider-pingdirectory](https://github.com/pingidentity/terraform-provider-pingdirectory) repository cloned
 - A GitHub Access Token. You can find instructions on how to do this [here](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic).
   - **The access token will only need repo, workflow, and read:org permissions**.
 - GitHub CLI Tool installed
   - This can be done using `brew` on Mac using the command `brew install gh`. For other OS, use the instructions listed [here](https://cli.github.com/manual/installation).
   
+## Important Note
+- The Pull Request to be included will need to be related to a specific issue branch (Example: 1-test-issue-to-be-resolved)
```
Comment (henryrecker-pingidentity) [inline contributing/release-notes.md:11]: Is there any way we could do this without this requirement? What if someone messes up and their PR didn't have the issue number after it's already been merged?

File: scripts/generate-changelog.sh (hunk 1 above)
Comment (wesleymccollam) [reply scripts/generate-changelog.sh:23]: Unsure currently how to handle multiple cases, other than just putting the issue in every category. I put a section in the README for this [here](https://github.com/pingidentity/terraform-provider-pingdirectory/commit/b525307c02f8307eecd606326b6f1ef427b6976c#diff-e31859efabfd97c974d26eed4f95d5c284345fbb674132a891933f7fb8bf999cR38).

File: contributing/release-notes.md (hunk 2 above)
Comment (wesleymccollam) [reply contributing/release-notes.md:11]: This is a big necessity for the script to work properly. I'm unsure of how you would relate an issue to a PR without having the corresponding branch this way. It also seems like a standard we should follow. This is how one of the other Ping Providers is handling this as well - example [here](https://github.com/pingidentity/terraform-provider-pingone/pull/283).

Comment (wesleymccollam) [reply contributing/release-notes.md:11]: We could also just manually include the issue in the CHANGELOG.md during the PR after it's generated if there are any issues.

**Review by henryrecker-pingidentity — APPROVED** [review]

**Pull request merged**